src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingArchery.java

```diff
@@ -284,6 +284,30 @@ public BuildingEntry getBuildingRegistryEntry()
         return ModBuildings.archery;
     }
 
+    @Override
+    public void onWakeUp()
+    {
+        super.onWakeUp();
+        
+        final World world = getColony().getWorld();
+        if (world == null)
+        {
+            return;
+        }
+
+        for (final BlockPos pos : bedList)
+        {
+            BlockState state = world.getBlockState(pos);
+            state = state.getBlock().getExtendedState(state, world, pos);
+            if (state.getBlock() instanceof BedBlock
+                  && state.get(BedBlock.OCCUPIED)
+                  && state.get(BedBlock.PART).equals(BedPart.HEAD))
+            {
+                world.setBlockState(pos, state.with(BedBlock.OCCUPIED, false), 0x03);
+            }
+        }
+    }    
```

Comment:
Is this not supposed to be default behaviour?
Why was this added here?

Reply:
All of our buildings where citizens sleep have this code. I missed porting it when I enabled sleeping in the Archery and the Combat Academy, which meant that the citizens would sleep one night, and go about there business the next day with the beds still marked occupied. The next night, they couldn't find a bed, so would stand around the hut sleeping.

Reply:
This will be solved in the future when we modularize this

Reply:
I got a PR going atm that will clean that up. Might need another week to be mature though